fleet/agent_group_service.go

```diff
@@ -121,6 +123,28 @@ func (svc fleetService) EditAgentGroup(ctx context.Context, token string, group
 	// append both lists and remove duplicates
 	// need to unsubscribe the agents who are no longer matching with the group
 	list := removeDuplicates(listSub, listUnsub)
+
+	// connect all agents to the group channel (check the already connected and connect the new ones)
+	if !reflect.DeepEqual(listSub, listUnsub) {
+		for _, a := range listSub {
```

Comment:
this will connect the new ones but don't we also have to disconnect the `listUnsub`?

Reply:
I haven't tested the disconnect feature. I'm not sure if it'll cause an error on the agent... or if disconnecting it's a graceful thing.